src/utils/pgwire/src/pg_protocol.rs

```diff
@@ -57,6 +61,29 @@ where
     unnamed_portal: Option<PgPortal<VS>>,
     named_statements: HashMap<String, PgStatement>,
     named_portals: HashMap<String, PgPortal<VS>>,
+
+    // Used for ssl connection.
+    tls_context: SslContext,
+}
+
+/// Configures TLS encryption for connections.
+#[derive(Debug, Clone)]
+pub struct TlsConfig {
+    /// The path to the TLS certificate.
+    pub cert: PathBuf,
+    /// The path to the TLS key.
+    pub key: PathBuf,
+}
+
+impl TlsConfig {
+    fn new_default() -> Self {
+        Self {
+            // Now the demo crt and key are hard code generated via simple self-signed CA.
+            // In future it should change to configure by user.
+            cert: "src/utils/pgwire/tests/ssl/demo.crt".into(),
+            key: "src/utils/pgwire/tests/ssl/demo.key".into(),
+        }
```

Comment:
What will the demo key and cert be used for?

Reply:
Used for SSL handshake. There must be certificate and private key.